### ISSUE TITLE
Cheap WiFi RGB LED bulbs and modules 

### ISSUE THREAD

**Issue by triphoppingman** [destil/MoodSync#5]
Hello there!

Fantastic tool.   Do you have any interest in me exploring this tool interfacing with the cheap RGB LED bulbs and modules like:

http://www.amazon.com/MagicLight%C2%AE-WiFi-LED-Light-Bulb/dp/B00SIDVZSW/ref=sr_1_sc_4?ie=UTF8&qid=1446469776&sr=8-4-spell&keywords=magicbulb

and:

http://www.amazon.com/gp/product/B00EUVDQL0?psc=1&redirect=true&ref_=oh_aui_detailpage_o08_s00

(I believe that these are all based on the ESP8266 chip but I have not cracked open the cases)

I have pretty much figured out the protocol (using python, see:  https://github.com/triphoppingman/rgbledcontrol.py).   But I see this android app (together with my NVidia Shield) as an awesome ambilight-lite project.

Let me know your thoughts.
Bruce.

**Comment by destil** [comment]
Hi, I'm not planning to myself but I would be grateful for a pull request from you.

**Comment by triphoppingman** [comment]
Ah - sounds great - I have actually managed to get this working (after a fashion) using some static address.   The only thing that I need some guidance on is how to integrate (if at all)?   I mean the tools is tied to LIFX but I am looking at tearing all of that out :)   Or do you think we should approach this as additional functionality?

**Comment by destil** [comment]
Hi, I think the UI can be the same and once the user clicks the ON button, it will start searching for all lights - LIFX, Hue, cheap RGB LED. The first light it founds wins and the screen color is mirrored there.

This is a good start - most people have just one type of lights so it will work. Future improvement could be some light selection UI.

You might need to refactor code a bit and introduce some abstractions so it's not that tight to LIFX. LIFX should be just one type of light it supports. So don't tear LIFX support if you want to submit a pull request.

**Comment by triphoppingman** [comment]
Yeah - that makes more sense to make it a general app.   I do have some concerns about the approach, I'd prefer to just enter an IP address (or select a discovered address) and then go from there.   I have a bunch of these for various applications and I'd rather not have this app stomp all over my other clients.

**Comment by destil** [comment]
IP address is too geeky, my app is more for general public. You just press one button and it works :) I would like to keep it that way. You can do some geeky settings hidden somewhere though.

**Comment by triphoppingman** [comment]
I hear you but in response I'd respectfully submit that people with RGB LEDs are most likely geeks.   Let me think on it.

**Comment by cumanzor** [comment]
For what is worth, the LIFX is at the same price as those cheap bulbs today.
